File: src/test/java/com/bruno/WordMatcherTest.java
package com.bruno;

import static org.junit.Assert.assertThat;
import static org.hamcrest.CoreMatchers.hasItems;

import com.bruno.matchers.Matcher;
import org.junit.BeforeClass;
import org.junit.Test;

public class WordMatcherTest {

    private static WordMatcher wordMatcher;

    @BeforeClass
    public static void setup() {
        wordMatcher = new WordMatcher(Language.GERMAN, Matcher.MISSING);
    }

    @Test
    public void should_find_words() throws Exception {
        assertThat(
                wordMatcher.getMatches("L_eb_"),
                hasItems("liebe", "liebt")
        );
    }

    @Test
    public void should_find_first_word() throws Exception {
        assertThat(
                wordMatcher.setLimit(1).getMatches("L_eb_"),
                hasItems("liebe")
        );
    }
}

File: src/test/java/com/bruno/WordMatcherTest.kt
package com.bruno

import org.junit.Assert.assertThat
import org.hamcrest.CoreMatchers.hasItems
import org.hamcrest.CoreMatchers.equalTo

import com.bruno.matchers.Matcher
import org.junit.Test

class WordMatcherTest {

    private val wordMatcher = WordMatcher(Language.GERMAN, Matcher.MISSING)

    @Test
    @Throws(Exception::class)
    fun should_find_words() {
        assertThat(
                wordMatcher.getMatches("L_eb_"),
                equalTo(listOf("liebe", "liebt"))
        )
    }

    @Test
    @Throws(Exception::class)
    fun should_find_first_word() {
        assertThat(
                wordMatcher.setLimit(1).getMatches("L_eb_"),
                equalTo(listOf("liebe"))
        )
    }
}
